### PR TITLE
Editor & Modal Updates

### PR DESCRIPTION
**Pull request by Parasrah**
# Fixes
- Added dependencies to `package.json`
- If you close the last file currently open, it will close the editor
- If you close a tab, a new tab will become active and the editor will display that file

# Features
- Editor will now auto-focus to code mirror instance when a file is opened, currently placing cursor at position 0, 0 in the file editor
- Pressing `enter` on the file creation modal will now default to `file` type

### CODE REVIEW

File: src/interface/tabs.js
Comment (t-mullen) [inline src/interface/tabs.js]: Better written as `tabs[index] || tabs[index - 1]`

File: src/interface/templates.js
Comment (t-mullen) [inline src/interface/templates.js:42]: Could you explain this change?

**Review by t-mullen — CHANGES_REQUESTED** [review]

File: src/interface/templates.js
Comment (Parasrah) [reply src/interface/templates.js:42]: This way the data type is considered a file. Because I added the selector for `.filename-input` to the modal `onKeyUp` listener, when you press enter on the new file input it treats it similarly to if you had clicked the new file button.

File: src/interface/tabs.js
Comment (Parasrah) [reply src/interface/tabs.js]: Good call! I'll make that change!

File: src/interface/interface.js
Comment (Parasrah) [inline src/interface/interface.js:14]: My formatter is being a ``${badWord}``. I can turn it off if you're attached to the spacing?

**Review by Parasrah — COMMENTED** [review]
@RationalCoding 
I fixed the issue you pointed out, and also fixed some of my formatting that broke standard

**Comment by Parasrah** [comment]
@RationalCoding waiting for approval? Anything that needs to be fixed?

**Review by t-mullen — APPROVED** [review]

**Pull request merged**
